### ISSUE TITLE
Cannot reproduce the results of AdaptSAM-huge

### ISSUE THREAD

**Issue by lostkevin** [YukunLi99/AdaptSAM#1]
I run the command in the README but only get the following results:

step=10000
| AP | AP50 | AP75 | APs | APm | APl |
| --- | --- | --- | --- | --- | --- |
| 5.17 | 8.02 | 5.75 | 4.17 | 7.51 | 7.11 |

The dataset I used is COCO2017 dataset w/o any modification. Could you please tell me how to reproduce the results in the table? Thanks!

**Comment by YukunLi99** [comment]
Thank you for your interest in our work! In the table, the results are presented using the metric of **Average Recall (AR)** instead of Average Precision (AP). Actually , [finetune_lora_sam_coco.py](https://github.com/YukunLi99/AdaptSAM/blob/main/configs/finetune/finetune_lora_sam_coco.py) is the configuration file used for COCO2017 dataset. If unable to reproduce the results, we will perform a thorough check of the configuration file to ensure its reproducibility.

**Comment by lostkevin** [comment]
I use the following command:
> python ./tools/lazyconfig_train_net.py --config-file configs/finetune/finetune_lora_sam_coco.py --num-gpus 8 

And I reduce the batch size to 8 due to the OOM (The learning rate is also halved). Will this cause the problem?

**Comment by lostkevin** [comment]
I guess the poor performance may result from the number of steps. I will try to double steps tonight. Thanks you!

**Comment by lostkevin** [comment]
step=20000

| $AP$ | $AR$ | $AR_{s}$ | $AR_{m}$ | $AR_{l}$ | 
| --- | --- | --- | --- | --- |
| 8.39 | 48.9 | 27.5 | 58.3 | 73.1 |

That looks OK! But I cannot figure out what $AP_{100}$ and $AR_{100}$ are. Can you give me some advice?

**Comment by YukunLi99** [comment]
For instance segmentation or object detection, AR (Average Recall) and AP (Average Precision) are evaluation metrics used to assess the performance of algorithms. Here's an explanation of the difference between AR and AP :

1. Average Recall (AR): AR measures the **overall recall** of an instance segmentation system across multiple intersection-over-union (IoU) thresholds. It evaluates how well the system detects and correctly localizes instances across a range of IoU thresholds. **AR is computed by calculating the recall (the ratio of true positives to the total number of ground truth instances) at different IoU thresholds and then averaging them.** Higher AR indicates better performance in terms of capturing more true positive instances across different IoU levels.
2. Average Precision (AP): AP measures **the precision-recall trade-off** of an instance segmentation system at a specific IoU threshold. Precision is the ratio of true positives to the total number of predicted instances, while recall is the ratio of true positives to the total number of ground truth instances. **AP is computed by calculating the area under the precision-recall curve at a specific IoU threshold.** It quantifies the system's ability to accurately segment instances at that particular IoU threshold. Higher AP values indicate better precision and recall trade-off at the given IoU threshold.

For a more detailed explanation, please refer to the following link :

- ​	https://jonathan-hui.medium.com/map-mean-average-precision-for-object-detection-45c121a31173
- ​	https://zhuanlan.zhihu.com/p/55575423

I hope this is useful to you.

**Comment by lostkevin** [comment]
So is $AP_{100}$ just $AP$ ? I mean as the 50 in $AP_{50}$ stands for the iou threshold 0.5, the iou threshold 1.0 seems strange in this way. Is it the mAP calculated with max 100 detections?

**Comment by Z-MU-Z** [comment]
Yes, here AP_100 is the Average Precision
(AP)[
IOU=0.50:0.95
area= all
maxDets=100 ] = 0.101 in the evaluation outputs

**Comment by Z-MU-Z** [comment]
I rerun python ./tools/lazyconfig_train_net.py --config-file configs/finetune/finetune_lora_sam_coco.py --num-gpus 4 on A100, the result is correct.

**Comment by lostkevin** [comment]
Thank you！
